marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -602,7 +606,7 @@ private void addToBatcher(File file, Format fileFormat) throws  IOException{
                     }
                     uri =  generateAndEncodeURI(outputURIReplace(uri)).replace("%", "%%");
                     Stream<DocumentWriteOperation> documentStream =  DocumentWriteOperation.from(
-                        contentStream, DocumentWriteOperation.uriMaker(uri +"/%s.json"));
+                        contentStream, DocumentWriteOperation.uriMaker(uri +"/%s." + ("xml".equalsIgnoreCase(outputFormat) ? "xml":"json")));
```

Comment:
For an easy unit test, could extract an e.g. "protected String buildUriForCsv(File)" method. Looks like most of the existing code could be tested very easily by a set of unit tests:

```
String uri = file.getParent();
                    if(SystemUtils.OS_NAME.toLowerCase().contains("windows")){
                        uri = "/" + FilenameUtils.separatorsToUnix(StringUtils.replaceOnce(uri, ":", ""));
                    }
                    uri =  generateAndEncodeURI(outputURIReplace(uri)).replace("%", "%%");
                    Stream<DocumentWriteOperation> documentStream =  DocumentWriteOperation.from(
                        contentStream, DocumentWriteOperation.uriMaker(uri +"/%s.json"));
```

Reply:
For unit tests, some more refactoring is required like "outputFormat" is set only when the flow is run. So, I am adding a functional test for now. I will add a task for better unit test coverage to be taken up post 5.0